feat: cognito auth for frontend and backend

gadomski opened the pull request:
It's hitting a custom app client at https://757082729759-xjxo3jyd.us-east-1.console.aws.amazon.com/cognito/v2/idp/user-pools/us-east-1_8VGoEWFWT/applications/app-clients/4jqhg58eag3h1bdlv5gj3hob1u/login-pages?region=us-east-1. Some notes about setup:

- You have to create at least one user before it'll work
- You have to create a managed login style before it'll work
- Right now I've got two redirect URLs in the app, but you can add more

We'll want to coordinate this with any sort of Terraform deploy to make sure they're in-sync.

https://github.com/user-attachments/assets/8870b54d-b511-4446-b7d2-42b36a9eec5e


## Checklist

<!-- Feel free to remove any items that don't apply to your PR (e.g. small fixes may not require documentation updates). -->

- [x] Pre-commit hooks pass (`uv run prek run --all-files`)
- [x] Documentation updated
- [x] Tests pass (`uv run pytest --integration`)
- [x] The PR's title is formatted per [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)

gadomski commented on frontend/src/main.tsx:
```diff
@@ -2,9 +2,22 @@ import { StrictMode } from 'react';
+    client_id: import.meta.env.VITE_COGNITO_CLIENT_ID,
+    redirect_uri: import.meta.env.DEV
+        ? 'http://localhost:5173'
+        : window.location.origin,
```
This was Claude-generated, not sure if it's the right way?

aliziel commented on backend/src/atlas_assistant/api.py:
```diff
@@ -8,59 +8,64 @@
 
 
-app = FastAPI(lifespan=lifespan)
+oidc = OpenIdConnect(openIdConnectUrl=settings.oidc_url)
```
Not a huge deal, but I wasn't able to auth through the `/docs` interface. 

Tried both logged in and out, but got: 
```
auth error {
    "error_description":"unauthorized_client",
    "state":"RnJpIE5vdiAyMSAyMDI1IDE3OjQwOjQ5IEdNVC0wODAwIChQYWNpZmljIFN0YW5kYXJkIFRpbWUp",
    "error":"invalid_request"
}
```

Just want to make sure this isn't unexpected for the changes as they're written. 

(Not advocating for more password handling or pulling secrets down either, the team can adjust based on their threat model)

gadomski replied:
What value did you use for clientId? It should be the id in https://757082729759-xjxo3jyd.us-east-1.console.aws.amazon.com/cognito/v2/idp/user-pools/us-east-1_8VGoEWFWT/applications/app-clients/4jqhg58eag3h1bdlv5gj3hob1u/quick-setup-guide?region=us-east-1.

But yeah, if it doesn't work now I don't think it's worth it to debug. Once we get the real cognito pool setup via terraform, then we'll chase down things to make sure they work for everyone.

aliziel replied:
>What value did you use for clientId?

The `4jqh` string, same values that were in the [.env.example](https://github.com/AdaptationAtlas/adaptation-atlas-assistant/pull/147#discussion_r2547087377) before I suggested more generic placeholders. Believe I double checked in console too.

Agreed though, better to check it out as part of the build. Tried to match the TF to what this is built against too, only thing thats different is the MFA requirement which wouldn't impact. Lets see if it recurs then

aliziel replied:
Reopening to confirm its happening with the new pool. Perhaps my browser is set differently than yours in the demo video, but my client session isn't available on the server host (no JWT in local storage, no cookies). I think there's a way to have that auth modal redirect to an external login, which I'll look int for the Cognito UI.

Also noting I added the local callbacks from your pool to the new build, and subsequent builds will overwrite. I'll look into local Cognito options so that doesn't have to be permanent

aliziel reviewed (approved):
Couple nits/clarifications, but LGTM

gadomski commented on backend/openapi.json:
```diff
@@ -750,14 +589,9 @@
-        }
+      "OpenIdConnect": {
+        "type": "openIdConnect",
+        "openIdConnectUrl": "https://cognito-idp.us-east-1.amazonaws.com/us-east-1_8VGoEWFWT/.well-known/openid-configuration"
```
It's true that this will vary depending on environment. This value won't be used in the frontend, so it's just noise. Removed it in 3de000e10866b8fdfae2e67c15661be1a51e3821.

gadomski commented:
This looks ready to rock, I just tested it against the cognito pool:

https://github.com/user-attachments/assets/baa815b4-b29b-4734-8509-85d7e225670d

I've updated the frontend deploy w/ the appropriate cognito outputs, and added those values to the dev Github Action environment. I'm marking this as ready to deploy, and we can try to roll it out maybe today?

gadomski commented:
Marking as draft until we land https://github.com/AdaptationAtlas/adaptation-atlas-assistant/pull/141

gadomski commented:
Added some light docs on logging in here: https://github.com/AdaptationAtlas/adaptation-atlas-assistant/wiki/Logging-in
